Breadcrumb

AMIRSOHAIL1998 opened the pull request:
# Chnages in PR:
- Added Headless Version of breadcrumps
- Added 5 different Examples how to use HeadlessBreadcrumbs.

ShreyDhyani commented on components/breadcrumbs/src/HeadlessBreadcrumbs.tsx:
```diff
@@ -0,0 +1,62 @@
+						}
+					>
+						{item}{" "}
+						{index < breadcrumbs.length - 1 ? (
```
change "" to null

AMIRSOHAIL1998 replied:
replced empty string with null

ShreyDhyani commented on components/breadcrumbs/src/HeadlessBreadcrumbs.tsx:
```diff
@@ -0,0 +1,62 @@
+								: {}
+						}
+					>
+						{item}{" "}
```
Is this { " " } needed? Can we add gap-x-1 or gap-x-2 to give eual spacing between all elements?

AMIRSOHAIL1998 replied:
it is auto added when save it no added because separate classes is given for styles

ShreyDhyani commented on components/breadcrumbs/index.ts:
```diff
@@ -0,0 +1 @@
+export { default as HeadlessBreadcrumbs } from "./src/HeadlessBreadcrumbs";
```
Can we add another Component here. Which will have following features:
- Import HeadlessBreadCrumb
- Have a default seperatorr(>)
- Accepts an array of strings/ReactNode

Have a starting home Icon
We can call it ReusejsBreadcrumbs

ShreyDhyani replied:
Also what is the behaviour of the number of `crumbs` are more. For example if we have 5 elements i the array. How will that be displayed.

Suggestions, If more elements than it is possible to be displayed in screen we have to hide elements and display .... in its place for eg:
Home > Link > ... > Sample

Please refer to the above example. This is a functional behavior so should be part of HeadlessBreadcrumb component.

AMIRSOHAIL1998 replied:
added ReusejsBreadcrums with default Home Icon and Separator ison also handle case for is there is more item in breadcrumbs then it can be managed with maxItems props.

ShreyDhyani commented:
@AMIRSOHAIL1998 Is this tested with different screen size?

AMIRSOHAIL1998 commented:
> @AMIRSOHAIL1998 Is this tested with different screen size?

> ` [Added example for collapsible breadcrumbs](/locoworks/reusejs-react/pull/77/commits/5636548a79898ae03a6ee8a871dc9704842a22a0)`

I haven't tested it yet, but I made some changes. Specifically, I modified the ReusejsComponent to be completely responsive, allowing users to have full control over styling and responsiveness. Additionally, I updated the styles for the headless example component to ensure it is responsive. Now, it's entirely up to the user to decide how they want to manage styling and responsiveness for the component.
